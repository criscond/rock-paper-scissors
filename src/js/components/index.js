import renderTheGame from './template';

export default renderTheGame;
