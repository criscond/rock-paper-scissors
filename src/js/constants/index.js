const LOCALSTORAGE = {
  CURRENT_GAME_COMPLEXITY: 'game-complexity',
  CURRENT_GAME_TYPE: 'game-type',
  RULES: 'game-rules'
}

export {
  LOCALSTORAGE
};
