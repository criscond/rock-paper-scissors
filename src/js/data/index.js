const complexity = [{
  id: 1,
  type: 'basic',
  noOfWeapons: 3,
  name: 'Three weapons'
}, {
  id: 2,
  type: 'advanced',
  noOfWeapons: 5,
  name: 'Five weapons',
},
];

const gameType = [{
  id: 1,
  type: 'human-vs-computer',
  name: 'Human vs Computer'
}, {
  id: 2,
  type: 'computer-vs-computer',
  name: 'Computer vs Computer ',
},
];

const weapons = [
  'Scissors',
  'Paper',
  'Rock',
  'Lizzard',
  'Spock',
];

const gameData = {
  name: 'Rock - Scissors - Paper',
  complexity,
  gameType,
  weapons,
};

const menus = [{
  name: 'GitHub',
  url: 'https://github.com/criscond/rock-paper-scissors',
  rel: 'noreferrer'
}, {
  name: 'WiKi',
  url: 'https://en.wikipedia.org/wiki/Rock-paper-scissors',
  rel: 'noreferrer'
}
];

const copyright = 'CrisCond';

export {
  gameData,
  menus,
  copyright,
};

