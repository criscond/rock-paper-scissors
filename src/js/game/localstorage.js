const saveData = (key, value) => localStorage.setItem(key, value);

const getSavedData = (key) => localStorage.getItem(key);

export {
  saveData,
  getSavedData,
}
