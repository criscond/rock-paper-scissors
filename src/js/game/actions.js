import { gameData } from '../data';
import { saveData } from '.';
import { LOCALSTORAGE } from '../constants';
import renderTheGame from '../components';

const { CURRENT_GAME_TYPE, CURRENT_GAME_COMPLEXITY } = LOCALSTORAGE;

const changeOpponents = (value) => {
  saveData(CURRENT_GAME_TYPE, value);
  renderTheGame();
}

const changeComplexity = (value) => {
  saveData(CURRENT_GAME_COMPLEXITY, value);
  renderTheGame();
}

const resetGame = () => {
  const { gameType, complexity } = gameData;
  saveData(CURRENT_GAME_TYPE, gameType[0].id);
  saveData(CURRENT_GAME_COMPLEXITY, complexity[0].id);

  renderTheGame();
}

export { changeComplexity, changeOpponents, resetGame } 
