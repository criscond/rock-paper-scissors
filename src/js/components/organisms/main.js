import { gameData } from '../../data';
import { createElement } from '../services';
import text from '../atoms/text';
import gameComplexitySection from '../molecules/complexity-game-section';
import simulationSection from '../molecules/simulation-section';
import scoreBoardSection from '../molecules/score-board-section';
import playersSection from '../molecules/players-section';
import resetGameSection from '../molecules/reset-game-section';
import weaponsSection from '../molecules/weapons-section';
import { getCurrentGameType } from '../../game';

const main = () => {
  const { name } = gameData;
  const currentGameType = getCurrentGameType();
  const element = createElement({ htmlTag: 'main' });
  element.appendChild(text({ htmlTag: 'h1', value: name }));
  element.appendChild(gameComplexitySection());
  element.appendChild(playersSection());
  element.appendChild(scoreBoardSection());
  currentGameType === 1
    ? element.appendChild(weaponsSection())
    : element.appendChild(simulationSection());
  element.appendChild(resetGameSection());

  return element;
}

export default main;
