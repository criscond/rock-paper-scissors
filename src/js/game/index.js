import { initiateTheGame } from './init'
import { changeComplexity, changeOpponents, resetGame } from './actions';
import { getSavedData, saveData } from './localstorage';
import { getCurrentComplexity, getCurrentGameType } from './rules';
import { makeTheChalenge } from './score';
import { getMaxNoOfWeapons, generateRandomWeaponId, getWeaponId, getWeaponName } from './weapons';

export {
  changeComplexity,
  changeOpponents,
  generateRandomWeaponId,
  getCurrentComplexity,
  getCurrentGameType,
  getMaxNoOfWeapons,
  getSavedData,
  getWeaponId,
  getWeaponName,
  initiateTheGame,
  makeTheChalenge,
  resetGame,
  saveData,
}
