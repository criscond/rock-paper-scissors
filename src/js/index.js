import renderTheGame from './components';
import { initiateTheGame } from './game'
import '../css/index.scss';

initiateTheGame();
renderTheGame();
