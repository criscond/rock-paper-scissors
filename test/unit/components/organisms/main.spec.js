import main from '../../../../src/js/components/organisms/main';
import { saveData } from '../../../../src/js/game/localstorage';
import { LOCALSTORAGE } from '../../../../src/js/constants';
const { CURRENT_GAME_TYPE } = LOCALSTORAGE;
describe('Main', () => {

  it('should create the main with weapons section', () => {
    const element = main();

    expect(element).toMatchSnapshot();
  });

  it('should create the main with simulation section', () => {
    saveData(CURRENT_GAME_TYPE, 2);
    const element = main();

    expect(element).toMatchSnapshot();
  });
});
