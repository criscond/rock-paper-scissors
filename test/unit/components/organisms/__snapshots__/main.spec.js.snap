// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should create the main with simulation section 1`] = `
<main>
  <h1
    class=""
  >
    Rock - Scissors - Paper
  </h1>
  <section>
    <span
      class=""
    >
      Game complexity: 
    </span>
    <select
      id="gameComplexity"
    >
      <option
        value="1"
      >
        Three weapons
      </option>
      <option
        value="2"
      >
        Five weapons
      </option>
    </select>
  </section>
  <section>
    <span
      class=""
    >
      Select the opponents: 
    </span>
    <select
      id="gameType"
    >
      <option
        value="1"
      >
        Human vs Computer
      </option>
      <option
        value="2"
      >
        Computer vs Computer 
      </option>
    </select>
  </section>
  <section
    class="column"
  >
    <span
      class="score"
      id="score"
    >
      -= Score Board =-
    </span>
  </section>
  <section
    class="column"
  >
    <button
      id="simulateGame"
      type="button"
    />
  </section>
  <section
    class="column"
  >
    <button
      id="resetGame"
      type="button"
    />
  </section>
</main>
`;

exports[`Main should create the main with weapons section 1`] = `
<main>
  <h1
    class=""
  >
    Rock - Scissors - Paper
  </h1>
  <section>
    <span
      class=""
    >
      Game complexity: 
    </span>
    <select
      id="gameComplexity"
    >
      <option
        value="1"
      >
        Three weapons
      </option>
      <option
        value="2"
      >
        Five weapons
      </option>
    </select>
  </section>
  <section>
    <span
      class=""
    >
      Select the opponents: 
    </span>
    <select
      id="gameType"
    >
      <option
        value="1"
      >
        Human vs Computer
      </option>
      <option
        value="2"
      >
        Computer vs Computer 
      </option>
    </select>
  </section>
  <section
    class="column"
  >
    <span
      class="score"
      id="score"
    >
      -= Score Board =-
    </span>
  </section>
  <section
    class="column"
  >
    <h2
      class=""
    >
      Choose your weapon
    </h2>
    <div
      class="weapons"
      role="listbox"
    >
      <div
        class="scissors"
        role="option"
      >
        <h3
          class=""
        >
          Scissors
        </h3>
      </div>
      <div
        class="paper"
        role="option"
      >
        <h3
          class=""
        >
          Paper
        </h3>
      </div>
      <div
        class="rock"
        role="option"
      >
        <h3
          class=""
        >
          Rock
        </h3>
      </div>
    </div>
  </section>
  <section
    class="column"
  >
    <button
      id="resetGame"
      type="button"
    />
  </section>
</main>
`;
